src/ReduceNoise/NonlocalMean.jl

```diff
@@ -116,18 +117,23 @@ function (f::NonlocalMean)(out::AbstractArray{<:NumberLike, 2},
     return out
 end
 
-function (f::NonlocalMean)(out::AbstractArray{<:AbstractRGB, 2},
-                           img::AbstractArray{<:AbstractRGB, 2})
+function (f::NonlocalMean)(out::AbstractArray{T, 2},
+                           img::AbstractArray{T, 2}) where T<: AbstractRGB
     cv_out = channelview(out)
     cv_img = channelview(img)
-    for i in 1:3
+    for i in 1:size(cv_img, 1)
         f(view(cv_out, i, :, :), view(cv_img, i, :, :))
     end
+    return out
 end
 
-(f::NonlocalMean)(out::AbstractArray{<:Color3, 2},
-                  img::AbstractArray{<:Color3, 2}) =
-    f(of_eltype(RGB, out), of_eltype(RGB, img))
+function (f::NonlocalMean)(out::AbstractArray{T, 2},
+                  img::AbstractArray{T, 2}) where T<:Color3
+    img = RGB.(img)
+    tmp = similar(img)
+    out .= f(tmp, img)
+    # f(of_eltype(RGB, out), of_eltype(RGB, img))
+end
```

Comment:
This works six months ago but now it doesn't, so I have to explicitly copy the data

@timholy do you happen to know what's the issue here? There are many `NaN`s in the result if `eltype(img) == Lab{Float32}` and breaks the test

https://github.com/johnnychen94/ImageNoise.jl/blob/55edbc43e7c002ddba97587be99678b3348a94b0/test/ReduceNoise/NonlocalMean.jl#L55-L59